File: ios/ShortDrama/Sources/Domain/Entities/Episode.swift
import Foundation

/// Domain entity representing an episode within a drama series.
struct Episode: Codable, Identifiable, Equatable, Sendable {
    let id: String
    let dramaId: String
    let title: String
    let episodeNumber: Int
    let videoUrl: String
    let duration: Int
    let thumbnailUrl: String
    let description: String?
    let createdAt: String
    let updatedAt: String

    var isPlayable: Bool {
        !videoUrl.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/model/Episode.kt
package com.djs66256.short_drama.domain.model

/**
 * Domain entity representing an episode of a short drama.
 * Fields align with the Backend Zod Schema.
 */
data class Episode(
    val id: String,
    val dramaId: String,
    val title: String,
    val episodeNumber: Int,
    val videoUrl: String,
    val duration: Int,
    val thumbnailUrl: String,
    val description: String,
    val createdAt: String,
    val updatedAt: String,
)
